crates/daphne-server/src/router/extractor.rs

```diff
@@ -1,30 +1,104 @@
 // Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
 // SPDX-License-Identifier: BSD-3-Clause
 
-use super::{AxumDapResponse, DaphneService};
+use std::io::Cursor;
+
 use axum::{
     async_trait,
-    body::HttpBody,
+    body::{Bytes, HttpBody},
     extract::{FromRequest, FromRequestParts, Path},
 };
 use daphne::{
     constants::DapMediaType,
     error::DapAbort,
     fatal_error,
-    messages::{request::DapRequestMeta, AggregationJobId, CollectionJobId, TaskId},
-    DapError, DapRequest, DapResource, DapVersion,
+    messages::{
+        AggregateShareReq, AggregationJobId, AggregationJobInitReq, CollectionJobId, CollectionReq,
+        Report, TaskId,
+    },
+    DapError, DapRequest, DapRequestMeta, DapResource, DapVersion,
 };
 use daphne_service_utils::{bearer_token::BearerToken, http_headers, metrics};
 use http::{header::CONTENT_TYPE, HeaderMap, Request};
+use prio::codec::ParameterizedDecode;
 use serde::Deserialize;
 
+use super::{AxumDapResponse, DaphneService};
+
+pub trait DecodeFromDapHttpBody: Sized {
+    fn decode_from_http_body(bytes: Bytes, meta: &DapRequestMeta) -> Result<Self, DapAbort>;
+}
+
+pub enum AggregationJobReq {
+    Init(AggregationJobInitReq),
+    // TODO: support continue requests
+}
```

Comment:
Do you mean `AggregationJobContinueReq`?

This is required for VDAFs that require more than one round of "preparation". This is the case for Poplar1, but we don't currently implement this. For Prio2, Prio3, Pine, and Mastic (to be implemented), we only need one. 

I'm betting that we're never going to need to implement Poplar1, in which case we won't need to support variants. My suggestion would be not write code in anticipation of this feature.

Alternatively, replace the TODO with a note:

```rust
pub enum AggregationJobReq {
    Init(AggregationJobInitReq),
    // NOTE We currently don't support multi-round VDAFs, hence why we don't support `AggregationJobContinueReq`.
}
```